rio/src/RIO/Process.hs

```diff
@@ -672,9 +673,21 @@ augmentPath dirs mpath =
 augmentPathMap :: [FilePath] -> EnvVars -> Either ProcessException EnvVars
 augmentPathMap dirs origEnv =
   do path <- augmentPath dirs mpath
-     return $ Map.insert "PATH" path origEnv
+     return $ Map.insert "PATH" path normEnv
   where
-    mpath = Map.lookup "PATH" origEnv
+    -- On windows, env vars are case insensitive.
+    -- We pick the "last" PATH variable and ensure
+    -- it's full uppercase.
+    -- See https://github.com/commercialhaskell/rio/issues/234
+    normEnv 
+      | isWindows = Map.foldrWithKey
```

Comment:
If you fancy the other way, it could also be:

```hs
Map.fromList $ map (first T.toUpper) $ Map.toList tm'
```

which is used `mkProcessContext`.

Factor it out?

Reply:
Yeah, that looks better to me. It's easier to understand, and it _might_ be more efficient (though that would need profiling to analyze, I could be wrong).

Factoring out would be great. thanks!